fs-storage/src/base_storage.rs

```diff
@@ -11,7 +11,13 @@ pub trait BaseStorage<K, V>: AsRef<BTreeMap<K, V>> {
     /// Check if the storage is up-to-date,
     /// i.e. that the internal mapping is consistent
     /// with the data in the filesystem.
-    fn is_storage_updated(&self) -> Result<bool>;
+    /// Returns:
+    /// - `Ok(true)` if the on-disk data is outdated compared to the in-memory data.
+    /// - `Ok(false)` if the on-disk data is up-to-date with the in-memory data.
+    /// If there is an error retrieving the file metadata (e.g., the file doesn't
+    /// exist), `Ok(true)` is returned, indicating that the on-disk data is outdated
+    /// or non-existent.
+    fn is_outdated(&self) -> Result<bool>;
```

Comment:
Returning `Ok(true)` if there is an error retrieving the file metadata might hide the fact that an error occurred. 

Would it be clearer to separate the cases where the data is outdated from when the file doesn't exist in the first place? We could return `Err(_)` if the file doesn't exist.